backend/os/file.go

```diff
@@ -275,3 +323,53 @@ func ensureDir(location vfs.Location) error {
 	}
 	return nil
 }
+
+// If cursor is not (0,0) will copy original file to a temp file,
+//opening its file descriptor to the current cursor position.
+//If cursor is (0,0), just begin writing to new temp file.
+//No need to copy original first.
+func (f *File) getInternalFile() (*os.File, error) {
+	// this is the use case of vfs.file
+	if f.useTempFile == false {
+		if f.file == nil {
+			finalFile, err := openOSFile(f.Path())
+			if err != nil {
+				return nil, err
+			}
+			f.file = finalFile
+		}
+		return f.file, nil
+	}
+	// this is the use case of vfs.tempFile
+	if f.tempFile == nil {
+		localTempFile, err := f.copyToLocalTempReader()
+		if err != nil {
+			return nil, err
+		}
+		f.tempFile = localTempFile
+	}
+
+	return f.tempFile, nil
+}
+
+func (f *File) copyToLocalTempReader() (*os.File, error) {
+	tmpFile, err := ioutil.TempFile("", fmt.Sprintf("%s.%d", f.Name(), time.Now().UnixNano()))
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err = f.openFile(); err != nil {
+		return nil, err
+	}
+	// todo: editing in place logic/appending logic
```

Comment:
Should these comments still be here, or is there a reason the todo should be handled after merge?

Reply:
It is part of a larger change that will likely be the next fix on this file, but too out of scope for the current fix.  Left the code there and commented instead of deleting it and rewriting it later.

Reply:
Might be good to log an issue related to this TODO

Reply:
im not sure how to do that, but I agree it would be a good thing to do

Reply:
Now that you added issue #42, can you just put that your today comment?  something like this should suffice:
```
// todo: editing in place logic/appending logic (see issue #42)
```